src/librustc_trans/trans/base.rs

```diff
@@ -3083,6 +3083,12 @@ pub fn trans_crate<'tcx>(analysis: ty::CrateAnalysis<'tcx>)
     let ty::CrateAnalysis { ty_cx: tcx, export_map, reachable, name, .. } = analysis;
     let krate = tcx.map.krate();
 
+    let check_overflow = if let Some(v) = tcx.sess.opts.debugging_opts.force_overflow_checks {
+        v
+    } else {
+        !attr::contains_name(&krate.config[], "ndebug")
```

Comment:
(this may need to be changed to be based on optimization level rather than `ndebug`)